scripts/deploy.sh

```diff
@@ -50,14 +50,23 @@ if [[ "${DEPLOY_AIO}" == "yes" ]]; then
   fi
   # force the deployment of haproxy for an AIO
   export DEPLOY_HAPROXY="yes"
-  if [[ ! -d /etc/openstack_deploy/ ]]; then
+  if [[ ! -e /openstack ]]; then
     ./scripts/bootstrap-aio.sh
     pushd ${RPCD_DIR}
-      for filename in $(find etc/openstack_deploy/ -type f -iname '*.yml'); do
-        if [[ ! -a "/${filename}" ]]; then
-          cp "${filename}" "/${filename}";
+      for relative_path in $(find etc/openstack_deploy/ -type f -iname '*.yml'); do
+        # If relative_path exists in /etc/openstack_deploy then it is merged
+        # into then repo version overriding the repo value when there is a
+        # conflict.
+        if [[ -e "/${relative_path}" ]]; then
+          # Output is written to the base file
+          #                         Base                 Override
+          ../scripts/update-yaml.py "${relative_path}"   "/${relative_path}";
```

Comment:
How does this account for overrides that RPC specifically put in place? The idea was that we could override the OSA defaults with out own, should we diverge. As I'm reading this, OSA wins, provided the user hasn't customized this.

I'm struggling with the same problem during upgrades for when users upgrade env.d files; I'm not sure a simple copy is going to work.